### PR TITLE
Fix Decoding interlaced grayscale png's

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Basically similar to #830. I missed one spot with the same issue. I have added more test images to cover those cases.

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
Nice work. I had trouble sourcing test images for this myself.
